internal/handler/flight.go

```diff
@@ -0,0 +1,49 @@
+package handler
+
+import (
+	"context"
+	"github.com/dilyara4949/flight-booking-api/internal/domain"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/response"
+	"github.com/gin-gonic/gin"
+	"log/slog"
+	"net/http"
+	"strconv"
+)
+
+type FlightService interface {
+	GetAll(ctx context.Context, page, pageSize int, available bool) ([]domain.Flight, error)
```

Comment:
Since we have filtration in place (pagination and availability) I would suggest renaming to smth like `GetFlights`, for semantic clarity. Same applies to repo, etc.